### PR TITLE
Add and Use win_unicode_console

### PR DESCRIPTION
**Pull request by guysv**
Before this commit, python 2 on windows would throw an encoding
exception when requested to print a unicode character. I copied some
code from the original IPython repo that handled this problem

### CODE REVIEW

**Comment by guysv** [comment]
I would really appreaciate some help with Travis :/ never used it before, and the exception in the last run seems like some configuration error rather then my fault.

File: .travis.yml [hunk 1]
```diff
@@ -2,14 +2,14 @@ language: python
 python:
     - 3.6
     - 3.5
-    - 3.4
-    - 3.3
-    - 2.7
```
Comment (minrk) [inline .travis.yml]: Since this is 5.x, it needs to keep testing in these versions of Python

File: setup.py [hunk 2]
```diff
@@ -76,14 +76,16 @@
 
 setuptools_args = {}
 install_requires = setuptools_args['install_requires'] = [
-    'jupyter_client',
-    'ipython',
-    'ipykernel', # bless IPython kernel for now
+    'jupyter_client<6.0.0',
+    'ipython<6.0.0',
+    'ipykernel<5.0.0', # bless IPython kernel for now
```
Comment (minrk) [inline setup.py]: These shouldn’t be pinned down. requires_python takes care of this.

Comment (guysv) [reply setup.py]: What versions of python should be compatible with this release? Technically, 5.x still works on the newer versions.

Comment (minrk) [reply setup.py]: 5.x should work with Python 2.7, ≥3.3. But there shouldn't be any need to pin these packages. Did you encounter a compatibility issue here? What prompted the pinning?

Comment (guysv) [reply setup.py]: (woops, pressed the wrong button)
jupyter_console requires promt_toolkit<2.0.0 but ipython requires a newer version, thus the pinning.
With the current setup.py I am able to install jupyter_console in python3.3+ so pinning the python version seems to me like limiting eligible targets for no good reason.

**Comment by guysv** [comment]
@minrk Let's drop 3.3 support?

File: setup.py (hunk 2 above)
Comment (minrk) [reply setup.py]: ok, then let's leave ipython pinned as a known compatibility issue, but not ipykernel or jupyter_client if there's no reason there.

**Comment by guysv** [comment]
@minrk Any news about merging this?

**Comment by Carreau** [comment]
Trying to revive this repo, as this targets the 5.x branch, no objections, but not sure there will be a release.

**Pull request merged**